### PR TITLE
Fix several ResourceWarning: unclosed file

### PR DESCRIPTION
**Pull request by BoboTiG**
A clean-up PR for resources leak fixes.

### CODE REVIEW

**Comment by gforsyth** [comment]
Hey @BoboTiG -- thanks for putting this in!  We package in `ply` from upstream to avoid resolving external dependencies -- any changes to `ply` should be submitted to the upstream repo (https://github.com/dabeaz/ply) and then we can pull them down from there.

**Comment by BoboTiG** [comment]
Good to hear. I will update the PR and send changes upstream.

**Comment by scopatz** [comment]
We should probably also update to the most recent version of ply that we vend :)

**Comment by gforsyth** [comment]
I think there's a `make` directive that pulls down the latest ply release...

**Comment by scopatz** [comment]
hmm interesting `make xonsh/ply` didn't seem to do anything for me.

File: xonsh/ply/example/BASIC/basic.py [hunk 1]
```diff
@@ -15,7 +15,8 @@
 # If a runtime error occurs, we bail out and enter
 # interactive mode below
 if len(sys.argv) == 2:
-    data = open(sys.argv[1]).read()
+    with open(sys.argv[1]) as f:
```
Comment (gforsyth) [inline xonsh/ply/example/BASIC/basic.py]: These changes should be reverted and made upstream.

File: xonsh/ply/example/BASIC/basiclog.py [hunk 2]
```diff
@@ -23,7 +23,8 @@
 # If a runtime error occurs, we bail out and enter
 # interactive mode below
 if len(sys.argv) == 2:
-    data = open(sys.argv[1]).read()
+    with open(sys.argv[1]) as f:
```
Comment (gforsyth) [inline xonsh/ply/example/BASIC/basiclog.py]: These changes should be reverted and made upstream.

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**